src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -495,13 +495,14 @@ internal static IEnumerable<TrainTestData> CrossValidationSplit(IHostEnvironment
         /// </summary>
         internal static void EnsureGroupPreservationColumn(IHostEnvironment env, ref IDataView data, ref string samplingKeyColumn, int? seed = null)
         {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("rand");
             // We need to handle two cases: if samplingKeyColumn is provided, we use hashJoin to
             // build a single hash of it. If it is not, we generate a random number.
-
             if (samplingKeyColumn == null)
             {
                 samplingKeyColumn = data.Schema.GetTempColumnName("SamplingKeyColumn");
-                data = new GenerateNumberTransform(env, data, samplingKeyColumn, (uint?)seed);
+                data = new GenerateNumberTransform(env, data, samplingKeyColumn, (uint?)(seed ?? host.Rand.Next()));
```

Comment:
Why fallback to `host.Rand.Next()` instead of the fixed seed given to the MLContext?

I think this is doing (1), then (4) from: https://github.com/dotnet/machinelearning/issues/4752#issuecomment-580865994

`TrainTestSplit` etc should listen to `MLContext` seed if not given by the user for the specific component.

> I think the precedence for PRNG seeds should be:
> 1. User seed specified for the component instance (or from parent component if hiding behind another)
> 2. User seed specified in `MLContext`
> 3. Default seed in component (unsure if we should remove these, might be a component-by-component discussion)
> 4. Random (and more random than current -- see https://github.com/dotnet/machinelearning/pull/4645#discussion_r365552609)

Reply:
MLContext seed isn't exposed, i.e. I cannot directly use mlContext.Seed. The seed for Host.Rand is set from MLContext though.